src/reconcile/rec_visibility.c

```diff
@@ -357,6 +398,58 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
             break;
```

Comment:
Does this work for checkpoint? Checkpoint doesn't work to the end of the chain.

Reply:
Yeah, this is definitely a problem.

Reply:
@agorrod @bvpvamsikrishna 
I had a chat with @quchenhao just then and it seems like the only reasonable thing to do is to walk the entire chain in checkpoint.

Is this ok as a first attempt? I believe we can probably refactor this later to do it one pass if we do something clever like blend this iteration with the history store insertion.

Reply:
I'm letting checkpoint iterate over the entire list now by removing the `break` at the bottom. It seems fine so far but I'm having a look to see if there might be some bad side effect that I'm missing.